### PR TITLE
[blenderbot P0] add PretrainedConfig and unit test

### PR DESCRIPTION
**Pull request by ZwhElliott**

### CODE REVIEW

File: paddlenlp/transformers/blenderbot/modeling.py [hunk 1]
```diff
@@ -143,7 +77,7 @@ def init_weights(self, layer):
                 layer.weight.set_value(
                     paddle.tensor.normal(
                         mean=0.0,
-                        std=self.init_std if hasattr(self, "init_std") else self.blenderbot.config["init_std"],
+                        std=self.init_std if hasattr(self, "init_std") else self.config.init_std,
```
Comment (sijunhe) [inline paddlenlp/transformers/blenderbot/modeling.py]:
```suggestion
                        std=self.config.init_std,
```

File: paddlenlp/transformers/blenderbot/modeling.py [hunk 2]
```diff
@@ -247,12 +162,11 @@ def forward(self, input_ids, attention_mask=None):
         hidden_states = inputs_embeds + inputs_embed_pos
         encoder_input = self.encoder_dropout(hidden_states)
 
-        if attention_mask is None:
```
Comment (sijunhe) [inline paddlenlp/transformers/blenderbot/modeling.py]: 这里应该是attention_mask is None的逻辑，如果用户没有传入attention_mask，我们才需要提供一个默认的

File: paddlenlp/transformers/blenderbot/modeling.py [hunk 3]
```diff
@@ -711,27 +500,38 @@ class :class:`paddle.nn.TransformerDecoder` for more information regarding ``cac
...
             dtype=self.blenderbot.shared.weight.dtype,
             is_bias=False,
         )
-        self.register_buffer(
-            "final_logits_bias",
-            paddle.zeros((1, self.blenderbot.config["vocab_size"]), dtype=paddle.get_default_dtype()),
-        )
+
+        if hasattr(self, "final_logits_bias"):
+            self.final_logits_bias = paddle.zeros((1, config.vocab_size), dtype=paddle.get_default_dtype())
```
Comment (sijunhe) [inline paddlenlp/transformers/blenderbot/modeling.py:531]: 这里添加的逻辑有什么特殊原因吗？

File: tests/transformers/blenderbot/test_modeling.py [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_conditiona_generation_model(*config_and_inputs)
+
+    def test_causal_lm_model(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_causal_lm_model(*config_and_inputs)
+
+    def test_inputs_embeds(self):
+        # Direct input embedding tokens is currently not supported
+        self.skipTest("Direct input embedding tokens is currently not supported")
```
Comment (sijunhe) [inline tests/transformers/blenderbot/test_modeling.py]: 这个可以删除

File: tests/transformers/blenderbot/test_modeling.py [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+            attention_mask=input_mask,
+        )
+        self.parent.assertEqual(result.shape, [self.batch_size, self.seq_length, self.vocab_size])
+
+
+class BlenderbotModelTest(ModelTesterMixin, unittest.TestCase):
+    base_model_class = BlenderbotModel
+    return_dict: bool = False
+    use_labels: bool = False
+    use_test_inputs_embeds: bool = True
```
Comment (sijunhe) [inline tests/transformers/blenderbot/test_modeling.py]: 这里设为false即可

**Review by sijunhe — COMMENTED** [review]
nice work.

File: paddlenlp/transformers/blenderbot/modeling.py (hunk 3 above)
Comment (ZwhElliott) [reply paddlenlp/transformers/blenderbot/modeling.py:531]: 这里会显示已有"final_logits_bias"参数所以register报错，所以就加了判断是否有该参数的逻辑。

File: paddlenlp/transformers/blenderbot/modeling.py [hunk 6]
```diff
@@ -251,8 +166,10 @@ def forward(self, input_ids, attention_mask=None):
             attention_mask = (
                 paddle.cast(input_ids == self.pad_token_id, dtype=paddle.get_default_dtype()).unsqueeze([1, 2]) * -1e4
             )
-            attention_mask.stop_gradient = True
+        else:
+            attention_mask = attention_mask.unsqueeze([1, 2]) * -1e4
```
Comment (ZwhElliott) [inline paddlenlp/transformers/blenderbot/modeling.py:170]: 如果有attention mask，那么也需要扩展一下

File: paddlenlp/transformers/blenderbot/modeling.py [hunk 7]
```diff
@@ -251,8 +166,10 @@ def forward(self, input_ids, attention_mask=None):
             attention_mask = (
                 paddle.cast(input_ids == self.pad_token_id, dtype=paddle.get_default_dtype()).unsqueeze([1, 2]) * -1e4
             )
-            attention_mask.stop_gradient = True
```
Comment (sijunhe) [inline paddlenlp/transformers/blenderbot/modeling.py:254]: stop gradient两边都需要

File: paddlenlp/transformers/blenderbot/modeling.py (hunk 3 above)
Comment (sijunhe) [reply paddlenlp/transformers/blenderbot/modeling.py:531]: ok

**Review by sijunhe — APPROVED** [review]
lgtm!
